proto_parser/src/proto_parser.cc

```diff
@@ -15,20 +15,71 @@
 */
 
 #include "proto_parser.h"
+#include "message_value.h"
+#include "primitive_value.h"
 
 namespace wireless_android_play_analytics {
-
-void ProtoParser::PopulateFields(int& prev_field_line, 
+  
+void ProtoParser::PopulateFields(UpperLayerInfo* prev_field_line, 
```

Comment:
The input params should be immutable params first then mutable params, so that 

    const google::protobuf::TextFormat::ParseInfoTree& tree,
    const google::protobuf::Message& message,
    int indent_count,
    ProtoValue* proto_value,
    UpperLayerInfo* upper_layer_info